frontend/online-shop/src/components/productsList/TableCards.tsx

```diff
@@ -0,0 +1,66 @@
+import { DescriptionCard } from "./DescriptionCard";
+import { useStyle } from "../../styles/productsList/TableCards";
+import { GridList, GridListTile } from "@material-ui/core";
+import { Grid } from "@material-ui/core";
+import { useState } from "react";
+import { useEffect } from "react";
+import { getAll } from "../../services/ProductService";
+
+export const TableCards = (props: any) => {
+  const style = useStyle();
+
+  const [width, setWidth] = useState(window.innerWidth);
+  const [columns, setColumns] = useState(4);
+  const [data, setData] = useState<any[]>([]);
+
+  useEffect(() => {
+    setWidth(window.innerWidth);
+    try {
+      getAll().then((result) => {
```

Comment:
Nice, that's the way to do it

Reply:
Or you could use `await` and catch the result in a const. Both ways work

Reply:
And also instead of catching the error you could use the `.catch()` (I think that is it) callback after the `.then`. So something like `getAll().then(()=>()).catch((error) => ...)`